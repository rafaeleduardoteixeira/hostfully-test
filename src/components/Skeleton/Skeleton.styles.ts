import styled from 'styled-components';

interface SkeletonProps {
  width?: string;
}

export const Container = styled.div<SkeletonProps>`
  width: ${(props) => props.width || '100%'};
  height: 100%;
  border-radius: 10px;
  animation: skeleton-loading 1s linear infinite alternate;
  @keyframes skeleton-loading {
    0% {
      background-color: hsl(200, 10%, 90%);
    }
    100% {
      background-color: hsl(200, 10%, 95%);
    }
  }
`;
