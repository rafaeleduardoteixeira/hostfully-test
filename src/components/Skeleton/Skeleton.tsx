import React from 'react';
import { Container } from './Skeleton.styles';

interface SkeletonProps {
  width?: string;
  height?: string;
  padding?: string;
}

export const Skeleton = (props: SkeletonProps) => {
  return <Container {...props} />;
};
