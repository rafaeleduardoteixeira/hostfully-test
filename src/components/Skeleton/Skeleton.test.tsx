import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { Skeleton } from './Skeleton';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  useNavigate: () => mockedUsedNavigate,
}));

describe('Skeleton', () => {
  test('Snapshot test', () => {
    const { asFragment } = render(<Skeleton />);
    expect(asFragment()).toMatchSnapshot();
  });
  test('Should render Skeleton', () => {
    render(<Skeleton />);
    const SkeletonElement = screen.getByTestId('Skeleton');
    expect(SkeletonElement).toBeInTheDocument();
  });
});
