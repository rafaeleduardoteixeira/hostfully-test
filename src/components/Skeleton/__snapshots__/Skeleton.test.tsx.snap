// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="sc-braxZu dBUbgW"
    data-testid="Skeleton"
  />
</DocumentFragment>
`;
